en/02_Developer_Guides/14_Files/05_File_Manipulation.md

```diff
@@ -0,0 +1,419 @@
+$store = Injector::inst()->get(AssetStore::class);
+$result = $store->setFromString('This is some file content', 'example-file.txt');
+
+// Save a database record that points to the stored file.
+// Note that we pull the file name from the result because the asset store might have renamed it.
+$dbFile = DBField::create_field('DBFile', $result);
+$fileRecord = File::create(['Name' => $result['Filename'], 'File' => $dbFile]);
+$fileRecord->write();
+```
+
+### Storage conflict resolution
+
+When storing new files, it's possible to determine the mechanism the backend should use when it encounters
+an existing file pattern. The conflict resolution to use can be passed into the third parameter of the
+above methods (after content and filename). The available constants are:
+
+| Constant                            | If an existing file is found then:  |
+| ----------------------------------- | ----------------------------------- |
+| `AssetStore::CONFLICT_EXCEPTION`    | An exception will be thrown         |
+| `AssetStore::CONFLICT_OVERWRITE`    | The existing file will be replaced  |
+| `AssetStore::CONFLICT_RENAME`       | The backend will choose a new name. |
+| `AssetStore::CONFLICT_USE_EXISTING` | The existing file will be used      |
+
+If no conflict resolution scheme is chosen, or an unsupported one is requested, then the backend will choose one.
+The default asset store supports each of these.
+
+The conflict resolution is passed in to the `config` argument like so:
+
+```php
+use SilverStripe\Assets\File;
+use SilverStripe\Assets\Storage\AssetStore;
+
+// Store a file named "example-file.txt".
+$fileRecord = File::create();
+$fileRecord->setFromString(
+    'This is some file content',
+    'example-file.txt',
+    // If a file with that name already exists, let the file store rename this one.
+    config: ['conflict' => AssetStore::CONFLICT_RENAME]
+);
+$fileRecord->write();
+```
+
+See [file storage](./file_storage/) for more details about the way files are stored.
+
+## Accessing files via PHP
+
+As with storage, there are also different ways of loading the content (or properties) of the file:
+
+| Method                   | Description                                                |
+| ------------------------ | ---------------------------------------------------------- |
+| `File::getStream`        | Will get an output stream of the file content              |
+| `File::getString`        | Gets the binary content                                    |
+| `File::getURL`           | Gets the URL for this resource. May or may not be absolute |
+| `File::getAbsoluteURL`   | Gets the absolute URL to this resource                     |
+| `File::getMimeType`      | Get the mime type of this file                             |
+| `File::getMetaData`      | Gets other metadata from the file as an array              |
+| `File::getFileType`      | Return the type of file for the given extension            |
```

Comment:
Mixture of verb forms

Reply:
That was already the wording being used - I'm not going to change it in this PR. The refactoring was just done so that there's a sensible place for the new documentation to live.